ADuCM3029_demo_cn0410: Initial commit

mirceacaprioru opened the pull request:
This is an initial comit of the CN0410 application code. This features the
EVAL-CN0410-ARDZ shield along with ADICUP3029. It uses a 16-bit unbuffered
voltage output DAC to control the voltage supplied to the LED-bar.

Signed-off-by: Mircea Caprioru <mircea.caprioru@analog.com>

stefpopa commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.cpp:
```diff
@@ -0,0 +1,216 @@
+	{
+		//Initialization for SPI and GPIO
+		ADI_SPI_RESULT eSpiResult;
+		int8_t i8Result;
```
you should probably initialize this variable with 0, otherwise when you return it the result might be unpredictable.

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+
+		u16RawValue = (uint16_t)atoi((char *)&Buffer[6]);
+		*u16DacValue = u16RawValue;//(uint32_t)(u16RawValue * 2)/2048 * 65535;
+		return 0;
```
add an extra blank line before returning

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+	{
+		return -1;
+	}
+	else
```
you can probably remove this else

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+		}
+
+		u16RawValue = (uint16_t)atoi((char *)&Buffer[6]);
+		*u16DacValue = u16RawValue;//(uint32_t)(u16RawValue * 2)/2048 * 65535;
```
commented code

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+
+	uint8_t u8Channel;
+	uint16_t  u16DacValue, u16ChannelValue;
+	uint8_t u8Error;
```
you can group the `u8Error `variable on the same line with `u8Channel`

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+	cn0410.SendCommand(AD5686_POWER, 0x00, 0x00); //normal power mode for all channels
+
```
do we need to call this four times?

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+
+		if ( u8Error == 0 )
+		{
+
```
extra blank line

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+
+	timer_sleep(10); //10 ms delay
+
+	cn0410.SendCommand(AD5686_RESET, 0x00, 0x00); //soft reset to zero scale
```
please check if this RESET command doesn't overwrite the values that you have configured before for internal reference and power mode

stefpopa commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.h:
```diff
@@ -0,0 +1,107 @@
+	};
+
+	enum ad5686_dac_channels
+	{
```
add a `AD5686_DAC_ALL = 0x00` element

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,232 @@
+				break;
+			}
+			}
+
```
we should add a default case which returns an "invalid value" or something

adrimbarean commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.cpp:
```diff
@@ -0,0 +1,216 @@
+		int8_t i8Result;
+
+		/* Initialize SPI driver */
+		if ((eSpiResult = adi_spi_Open(CN0410_SPI_DEV_CFG , m_spi_memory, ADI_SPI_MEMORY_SIZE, &m_spi_handle)) != ADI_SPI_SUCCESS)
```
If we want to use Linux coding style lines should be equal or shorter than 80 columns and go beyond only when it improves readability. Also nitpick: extra space in between if and condition.

adrimbarean commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.cpp:
```diff
@@ -0,0 +1,216 @@
+		if ((eSpiResult = adi_spi_Open(CN0410_SPI_DEV_CFG , m_spi_memory, ADI_SPI_MEMORY_SIZE, &m_spi_handle)) != ADI_SPI_SUCCESS)
+		{
+			i8Result = -1;
+		}
```
Open bracket should be on the same line with the if and separated by space. But in this case maybe brackets are unnecessary because the if guards only one statement.

adrimbarean commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.cpp:
```diff
@@ -0,0 +1,216 @@
+		adi_gpio_SetHigh(LDAC_PORT, LDAC_PIN);
+		adi_gpio_SetLow(LDAC_PORT, LDAC_PIN);
+		adi_gpio_SetHigh(LDAC_PORT, LDAC_PIN);
+
```
Nitpick: an extra line.

adrimbarean commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.h:
```diff
@@ -0,0 +1,107 @@
+		ADI_SPI_HANDLE      m_spi_handle;
+		/*!< SPI transceiver buffer.         */
+		ADI_SPI_TRANSCEIVER m_transceive;
+
```
Nitpick: extra line.

mirceacaprioru commented:
Changelog V1 -> V2

- removed extra blank lines
- ran astyle on the project files
- removed multiple calls to cn0410.SendCommand(AD5686_POWER,	AD5686_DAC_NONE, 0x00);
- removed switch brackets

commodo commented on projects/ADuCM3029_demo_cn0410/RTE/Device/ADuCM3029/ADuCM3029.ld:
```diff
@@ -0,0 +1,217 @@
+/* 
```
this file looks generated;
should it be here ?

mirceacaprioru replied:
The file is a script to configure the memory regions it applies to the controller used and is part of the RTE CMSIS platform folder. I would not remove this and keep the project independent from other packages.

mirceacaprioru commented:
if there are no other comments, will merge later today.

stefpopa commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.cpp:
```diff
@@ -0,0 +1,204 @@
+
+	int8_t CN0410::Init()
+	{
+		//Initialization for SPI and GPIO
```
please use only one type of comment: `/* */`  is preferred. Check this in all your code.

stefpopa commented on projects/ADuCM3029_demo_cn0410/Sensors/adi_cn0410.h:
```diff
@@ -0,0 +1,106 @@
+};
+
+enum ad5686_dac_channels {
+	AD5686_DAC_NONE = 0x00,
```
is it `AD5686_DAC_NONE `or `AD5686_DAC_ALL `?

mirceacaprioru replied:
There are some commands that do not require a specific channel to be selected. Since a transmit message has a channel selection section I created this enum value to use when this value does not matter.

stefpopa replied:
ok, thank you for clarifying.

stefpopa commented on projects/ADuCM3029_demo_cn0410/src/ADuCM3029_demo_cn0410.cpp:
```diff
@@ -0,0 +1,221 @@
+		*u8Channel = Buffer[4]; /*channel letter*/
+	} else {
+		*u8Channel = '0';
+	}
```
we don't need brackets here

mirceacaprioru commented:
Changelog V2 -> V3

- changed comments from // to /* */
- removed extra brackets for if statement
